src/main/java/com/google/api/codegen/gapic/GapicProvider.java

```diff
@@ -12,34 +12,31 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package com.google.api.codegen;
+package com.google.api.codegen.gapic;
 
-import com.google.api.tools.framework.model.Model;
-import com.google.common.collect.Multimap;
+import com.google.api.codegen.GeneratedResult;
 
-import java.io.IOException;
+import java.util.List;
 
 /**
  * A GapicProvider performs code or fragment generation using on a proto-based Model for a
  * particular language.
  */
 public interface GapicProvider<InputElementT> {
 
-  Model getModel();
-
   /**
-   * Gets the view of the model.
+   * Returns the snippet files that this provider will use for code generation.
    */
-  public InputElementView<InputElementT> getView();
+  List<String> getSnippetFileNames();
 
   /**
-   * Runs code generation.
+   * Runs code generation and puts the output in outputPath.
    */
-  GeneratedResult generate(InputElementT element, SnippetDescriptor snippetDescriptor);
+  void generate(String outputPath) throws Exception;
```

Comment:
Just a thought: this is not worse that what we had before, but it seems strange that a GapicProvider can implement arbitrary logic to generate and output, instead of creating an iterable of GeneratedResult (where the filename is set correctly), so that the XXXTool can do the outputting in a consistent manner. (I think usually the XXXTool class is intended to handle the direct I/O anyway?)

We don't need to change this now, though.